drivers/base/bootvid/i386/xbox/bootvid.c

```diff
@@ -430,7 +430,7 @@ VidScreenToBufferBlt(
         PUCHAR Buf = Buffer + y * Delta;
 
         /* Start the X inner loop */
-        for (ULONG x = 0; x < Width; x += 2)
+        for (ULONG x = 0; x < Width + 1; x += sizeof(USHORT))
```

Comment:
Why is the loop going from 0 up to "Width" included ? Shouldn't it be from 0 to Width-1 included (aka. to Width excluded) ?

Reply:
Because the loop definrd inside function iterates from `0` to `Width+1` https://git.reactos.org/?p=reactos.git;a=blob;f=drivers/base/bootvid/common.c;h=7979ea42a0481fed85baf98c6a80791ba44a0aac;hb=dd6eb40381be1a090e4130e6d5cdd9fea141aac9#l84

Reply:
Well, it's obviously wrong (buggy), because suppose that you specify a BitBlt with an image of width == 0, then you would expect the code to **NOT** touch any pixels, yet your rewrite (and that old do-while loop) does modify the pixels.